flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/ResultPartitionTest.java

```diff
@@ -205,6 +216,66 @@ protected void testAddOnPartition(final ResultPartitionType pipelined)
 		}
 	}
 
+	@Test
+	public void testReleaseMemoryOnBlockingPartition() throws Exception {
+		testReleaseMemory(ResultPartitionType.BLOCKING);
+	}
+
+	@Test
+	public void testReleaseMemoryOnPipelinedPartition() throws Exception {
+		testReleaseMemory(ResultPartitionType.PIPELINED);
+	}
+
+	/**
+	 * Tests {@link ResultPartition#releaseMemory(int)} on a working partition.
+	 *
+	 * @param resultPartitionType the result partition type to set up
+	 */
+	private void testReleaseMemory(final ResultPartitionType resultPartitionType) throws Exception {
+		final int numBuffers = 10;
+		final NetworkEnvironment network = new NetworkEnvironment(
+			new NetworkBufferPool(numBuffers, 128),
+			new LocalConnectionManager(),
+			new ResultPartitionManager(),
+			new TaskEventDispatcher(),
+			new KvStateRegistry(),
+			null,
+			null,
+			IOManager.IOMode.SYNC,
+			0,
+			0,
+			2,
+			8,
+			true);
+		final ResultPartitionConsumableNotifier notifier = mock(ResultPartitionConsumableNotifier.class);
+		final ResultPartition resultPartition = spy(createPartition(notifier, resultPartitionType, false));
```

Comment:
Same here. Do not use mockito for that kind of stuff. Instead of checking whether some exact method was called or not, and how many times with what arguments (those things can change while the code still remain valid while the mockito tests will start failing), check the state of the `NetworkBufferPool` - how many buffers are available after some kind of operation etc.